core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -170,6 +170,13 @@ class KafkaApis(val requestChannel: RequestChannel,
     val response =
       if (authorize(request.session, ClusterAction, Resource.ClusterResource)) {
         val (result, error) = replicaManager.stopReplicas(stopReplicaRequest)
+        result.map{case (topicPartition, errorCode) => {
```

Comment:
Hmm, it seems that before a broker receives a StopReplicaRequest, it must have already received a LeaderAndIsrRequest to become follower (except for the topic deletion case). In ```KafkaApis.handleLeaderAndIsrRequest``` we have already called ```coordinator.handleGroupEmigration()```. So it seems that we don't need to remove group again here.